torch/testing/_internal/common_device_type.py

```diff
@@ -377,6 +377,13 @@ def instantiated_test(self, param_kwargs=param_kwargs):
                 except RuntimeError as rte:
                     # check if rte should stop entire test suite.
                     self._stop_test_suite = self._should_stop_test_suite()
+                    # Check if test has been decorated with `@expectedFailure`
+                    # Based on https://github.com/python/cpython/blob/ffa505b580464d9d90c29e69bd4db8c52275280a/Lib/unittest/case.py#L164
+                    # Suppress the exception in that case
```

Comment:
This approach seems reasonable but it'd be nice if this comment was elaborated to explain how suppressing the exception like this ultimately results in its being thrown and associated with the test